text/0118-search-api.md

```diff
@@ -108,156 +111,273 @@ It returns the number of documents matching the current search query for each gi
+If `"*"` is provided as a value: `attributesToCrop=["*"]` all the attributes present in `attributesToRetrieve` will be automatically assigned to `attributesToCrop`.
 
-- 🔴 Sending a value with a different type than `Array[String]`(POST), `String`(GET) or `null` for `attributesToCrop` will return a [bad_request](0061-error-format-and-definitions.md#bad_request) error.
+- 🔴 Sending a value with a different type than `Array[String]`(POST), `String`(GET) or `null` for `attributesToCrop` returns a [bad_request](0061-error-format-and-definitions.md#bad_request) error.
 
-> See [_Formatted Field Behavior](0039-_formatted-field-behavior_.md)
+##### 3.1.11.1. Custom `cropLength` Defined Per Attribute.
 
-##### 1.2.1.10 `cropLength`
+Optionally, indicating a custom crop length for any of the listed attributes is possible:
+
+`"attributesToCrop":["attributeNameA:15", "attributeNameB:30"]`
+
+A custom crop length set in this way has priority over the `cropLength` parameter.
+
+##### 3.1.11.2. searchableAttributes
+
+As for `attributesToHighlight`, attributes not defined in the `searchableAttributes` index setting are also cropped if assigned to `attributesToCrop`.
+
+#### 3.1.1.12. `cropLength`
 
 - Type: Integer
 - Required: False
-- Default: `200`
+- Default: `10`
 
-Configures the number of characters to keep on each side of the matching query term when using the `attributesToCrop` parameter.
+Sets the total number of **words** to keep around the matched part of an attribute specified in the `attributesToCrop` parameter.
 
 If `attributesToCrop` is not configured, `cropLength` has no effect on the returned results.
 
-- 🔴 Sending a value with a different type than `Integer` or `null` for `cropLength` will return a [bad_request](0061-error-format-and-definitions.md#bad_request) error.
+- 🔴 Sending a value with a different type than `Integer` or `null` for `cropLength` returns a [bad_request](0061-error-format-and-definitions.md#bad_request) error.
+
+#### 3.1.1.13. `cropMarker`
+
+- Type: String
+- Required: False
+- Default: `"…"` (U+2026)
+
+Sets the crop marker to apply before and/or after cropped part selected within an attribute defined in `attributesToCrop`. See [3.1.11. `attributesToCrop`](#3111-attributestocrop) section.
+
+The specified crop marker is applied by following rules. See [3.1.1.13.1. Applying `cropMarker`](#311131-applying-cropmarker) section.
+
+Specifying `cropMarker` to `""` or `null` implies that no marker will be applied to the cropped part, if any.
 
-##### 1.2.1.11 `matches`
+- 🔴 Sending a value with a different type than `String` or `null` for `cropMarker` returns a [bad_request](0061-error-format-and-definitions.md#bad_request) error.
+
+##### 3.1.1.13.1. Applying `cropMarker`
+
+###### 3.1.1.13.1.1. Matched Part To Be Cropped
+
+The cropping algorithm tries to match the window with the highest density of query terms within the `cropLength` limit. Then it will pick the window that contains the more ordered query terms.
```

Comment:
Not sure to understand, especially the "Then it will pick the window that contains the more ordered query terms". Maybe an example would help :)

Reply:
I have rewritten this section "Matched Part To Be Cropped" with a detail of the density calculation. I hope it is clearer. 🙏